src/js/core/history/manager.js

```diff
@@ -130,17 +131,19 @@
 							options.link = link;
 						}
 						history.disableVolatileMode();
+						// mark as handled for back button
```

Comment:
code in if's statements seems to be uplicated
can we just write
if (rel === "back" || !triggerStateChange(options))
{
....
}
moreover we will avoid calling the same code twice when both conditions are passed

Reply:
Yes you are right.

Reply:
however, this does not work ...
Method `triggerStateChange` has argument `options` with `event`.
This event has to be prevented before call `triggerStateChange` method.

and `triggerStateChange` method has to be called.